src/dmd/dtemplate.d

```diff
@@ -1351,8 +1351,37 @@ extern (C++) final class TemplateDeclaration : ScopeDsymbol
             {
                 Parameter fparam = fparameters[parami];
 
+                /* https://issues.dlang.org/show_bug.cgi?id=16486
+                 * If 'prmtype' is template alias, resolve it,
+                 * to the instance it's aliased to before proceeding.
+                 */
+                Type prmtype = fparam.type;
+                //printf("prmtype: %s\n", prmtype.toChars());
+                while (prmtype.ty == Tinstance)
+                {
+                    TypeInstance prmti = cast(TypeInstance) prmtype;
```

Comment:
you can probably do `while (TypeInstance prmti = prmtype.isInstance())` (if `isInstance()` doesn't exist please create it) similar to the `if (auto td = prmti.tempinst.tempdecl.isTemplateDeclaration())` below.

Reply:
There is the isTemplateInstance but that is for symbols. If I understand it correctly, code will look kind of weird if I introduce that.

Reply:
`isTypeInstance()`?
https://github.com/dlang/dmd/blob/9585ab8535a4aff9861f89798ae5fc730778bf80/src/dmd/mtype.d#L2663

Reply:
That's what I needed, thanks

Reply:
Ah, thats what it is. Sorry about that, I'm going to blame travelling and jet lag for that.

Reply:
Don't mention it. It's odd though, because I saw the idiom I used in a lot of places in the compiler and not once the isTypeInstance(). It might be a recent addition...

Reply:
Do you still want that? Because I think it can't happen because it is a `while` and not an `if`.

Reply:
Yes:

```
if (TypeInstance prmti = prmtype. isTypeInstance())
{
    Objects* tempargs = prmti.tempinst.tiargs;
    while ((prmti = prmtype. isTypeInstance()))
    {
        //...
    }
}

```